v-next/hardhat-node-test-runner/src/protocol.ts

```diff
@@ -0,0 +1,148 @@
+import type { HardhatRuntimeEnvironment } from "hardhat/types/hre";
+
+import fs from "node:fs";
+import net from "node:net";
+import * as os from "node:os";
+
+import { assertHardhatInvariant } from "@nomicfoundation/hardhat-errors";
+
+function getSocketPath(): string {
```

Comment:
I think we need to generate a unique name, and pass it to the child processes somehow. Otherwise you wouldn't be able to run multiple instances of HH

Reply:
In what scenario wouldn't it work ? you mean running the tests with concurrency > 1 ?